gapis/api/vulkan/frame_loop.go

```diff
@@ -1470,6 +1470,17 @@ func (f *frameLoop) resetBufferViews(ctx context.Context, stateBuilder *stateBui
 
 func (f *frameLoop) resetImages(ctx context.Context, stateBuilder *stateBuilder) error {
 
+	for toDestroy := range f.imageToDestroy {
+		log.D(ctx, "Destroy image %v which was created during loop.", toDestroy)
+		image := GetState(f.loopEndState).Images().Get(toDestroy)
+		stateBuilder.write(stateBuilder.cb.VkDestroyImage(image.Device(), toDestroy, memory.Nullptr))
+
+		imageViewUsers := image.Views().All()
+		for imageView := range imageViewUsers {
+			f.imageViewToDestroy[imageView] = true
```

Comment:
Not needed here. You need to add all the *created* images' imageViews to both the imageViewToDestroy and imageViewToCreate maps below. I'll add a comment in the appropriate place.